File: src/web/AjaxGetMajors.java
package web;

import service.MajorService;
import service.impl.MajorServiceImpl;

import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.util.List;

@WebServlet("/servlet/AjaxGetMajors")
public class AjaxGetMajors extends HttpServlet {

    @Override
    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
        MajorService majorService = new MajorServiceImpl();
        StringBuilder result = new StringBuilder();
        for (String major : majorService.findMajors()) {
            result.append("<option>").append(major).append("</option>");
        }
        resp.getWriter().write(result.toString());
    }

}

File: src/web/AjaxGetMajors.kt
package web

import service.impl.MajorServiceImpl
import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@WebServlet("/servlet/AjaxGetMajors")
class AjaxGetMajors : HttpServlet() {

    override fun doPost(req: HttpServletRequest?, resp: HttpServletResponse?) {
        val majorService = MajorServiceImpl()
        var result = ""
        majorService.findMajors()?.forEach { result += "<option>$it</option>" }
        resp?.writer?.write(result)
    }

}